src/elements/content-sidebar/activity-feed/task-new/Task.js

```diff
@@ -0,0 +1,166 @@
+// @flow
+import * as React from 'react';
+import noop from 'lodash/noop';
+import { FormattedTime, FormattedMessage } from 'react-intl';
+import classNames from 'classnames';
+import { fillUserPlaceholder } from '../../../../utils/fields';
+import messages from '../../../common/messages';
+import { TASK_NEW_APPROVED, TASK_NEW_REJECTED, TASK_NEW_INCOMPLETE, TASK_NEW_COMPLETED } from '../../../../constants';
+import Comment from '../comment';
+import AssigneeStatus from './AssigneeStatus';
+import TaskActions from './TaskActions';
+import Status from './TaskStatus';
+import './Task.scss';
+
+type Props = {
+    ...TaskNew,
+    currentUser: User,
+    error?: ActionItemError,
+    getAvatarUrl: string => Promise<?string>,
+    getMentionWithQuery?: Function,
+    getUserProfileUrl?: string => Promise<string>,
+    isDisabled?: boolean,
+    isPending?: boolean,
+    mentionSelectorContacts?: SelectorItems,
+    onAssignmentUpdate: Function,
+    onDelete?: Function,
+    onEdit?: Function,
+    translatedTaggedMessage?: string,
+    translations?: Translations,
+};
+
+const MAX_AVATARS = 3;
+
+// eslint-disable-next-line react/prefer-stateless-function
+class Task extends React.Component<Props> {
+    render(): React.Node {
+        const {
+            assigned_to,
+            created_at,
+            created_by,
+            currentUser,
+            due_at,
+            error,
+            id,
+            isPending,
+            onDelete,
+            onEdit,
+            onAssignmentUpdate = noop,
+            name,
+            status,
+            translatedTaggedMessage,
+            translations,
+            getAvatarUrl,
+            getUserProfileUrl,
+            getMentionWithQuery,
+            mentionSelectorContacts,
+        } = this.props;
+
+        // Enable when edit/delete are implemented for new tasks
+        const taskPermissions = {
+            can_edit: false,
+            can_delete: false,
+        };
+
+        // // find assignment for current user if permitted
+
+        const currentUserAssignment =
+            assigned_to && assigned_to.entries
+                ? assigned_to.entries.find(({ target }) => target.id === currentUser.id)
+                : null;
+
+        const assigneeCount = (assigned_to && assigned_to.entries.length) || 0;
```

Comment:
It looks like we're checking the existence of `assigned_to` and `assigned_to.entries` twice. Can we normalize it into an array once above both checks?